R/scale-discrete-.R

```diff
@@ -140,6 +140,10 @@ ScaleDiscretePosition <- ggproto("ScaleDiscretePosition", ScaleDiscrete,
 
   map = function(self, x, limits = self$get_limits()) {
     if (is.discrete(x)) {
+      # Guard against old use of having identity palettes
+      if (identical(.subset2(self, "palette"), identity)) {
+        self$palette <- seq_len
+      }
```

Comment:
Could we fix this during construction? instead of checking at every map call?

Reply:
Good call 👍